src/figuresClasses.ts

```diff
@@ -1,11 +1,89 @@
-export interface Figure {}
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
 
-export class Triangle implements Figure {}
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
```

Comment:
[CODE KNOWLEDGE] - When declaring constructor parameters with access modifiers in TypeScript, there is no need to assign them manually to class properties as they are automatically created. You can remove the manual assignments from the constructor.

GOOD EXAMPLE:
constructor(public color: Color, public a: number, public b: number, public c: number) {
  // constructor logic
}